### DISCUSSION TITLE
ZBM version v2.3.0 Zfs version 2.2.0

### DISCUSSION THREAD

**Discussion by Sithuk** [zbm-dev/zfsbootmenu#608] [General]
I ran a test using the latest released ZBM (v.2.3.0) and Ubuntu 22.04. ZBM started on first boot. but was unable to import the root pool. I tried a "zpool import -f rpool" from the emergency shell and got the following. Ubuntu 22.04 installs zfs version 2.2.0, which I thought ZBM supported? Does anyone know what the problem might be?

"This pool uses the following feature(s) not supported by this system:
com.klarasystems:vdev_zaps_v2
cannot import 'rpool' : unsupported version or feature"

"Zfs version" in the emergency shell gives the following.
zfs-2.1.5-1ubuntu6\~22.04.3
zfs-kmod-2.1.5-1ubuntu6\~22.04.2

**Comment by ahesford** [comment]
ZFSBootMenu will import pools with unsupported features if those features are read-only compatible. OpenZFS 2.2 introduced several features that are not compatible at all, which is one of the reasons our guides recommend sticking with v2.1 compatibility flags unless you both require the new features and will make sure that your ZBM image supports them.

Our pre-built binary images use OpenZFS 2.2, but the version you report suggests that you are using a custom image from an older Ubuntu.

**Reply by Sithuk** [reply]
I used the source code from the following tarball to build zbm. I think that is the latest?
https://api.github.com/repos/zbm-dev/zfsbootmenu/tarball/v2.3.0

**Reply by zdykstra** [reply]
ZFSBootMenu source releases don't ship with ZFS. So whatever system you built it on has ZFS 2.1.5 installed.

**Comment by Sithuk** [comment]
Thank you zdykstra, I suspect I know what the issue is. I used an Ubuntu 23.10 live iso to install the current lts (22.04), which is of course an older version of Ubuntu. That'll be it.
